sympy/physics/units/tests/test_quantities.py

```diff
@@ -239,10 +241,30 @@ def test_sum_of_incompatible_quantities():
 
 
 def test_factor_and_dimension():
+    assert (3000, Dimension(1)) == Quantity._collect_factor_and_dimension(3000)
     assert (1001, length) == Quantity._collect_factor_and_dimension(meter + km)
     assert (2, length/time) == Quantity._collect_factor_and_dimension(
         meter/second + 36*km/(10*hour))
 
     x, y = symbols('x y')
     assert (x + y/100, length) == Quantity._collect_factor_and_dimension(
         x*m + y*centimeter)
+
+    cH = Quantity('cH', amount_of_substance/volume)
+    pH = -log(cH)
+
+    assert (1, volume/amount_of_substance) == Quantity._collect_factor_and_dimension(
+        exp(pH))
+
+    v_w1 = Quantity('v_w1', length/time, 1.5*meter/second)
+    v_w2 = Quantity('v_w2', length/time, 2*meter/second)
+    expr = Abs(v_w1/2 - v_w2)
+    assert (1.25000000000000, length/time) == \
+        Quantity._collect_factor_and_dimension(expr)
+
+    expr = v_w1 - Abs(v_w1)
+    assert (0, Dimension(1)) == Quantity._collect_factor_and_dimension(expr)
```

Comment:
Yes it should, and this is what actually happens, so the test is probably wrong:
```
expr = v_w1 - Abs(v_w2)
Quantity._collect_factor_and_dimension(expr)

(-0.500000000000000, Dimension(length/time))
```